src/WebJobs.Script.WebHost/Management/InstanceManager.cs

```diff
@@ -334,13 +334,29 @@ private async Task ApplyContext(HostAssignmentContext assignmentContext)
                         _logger.LogWarning("App is configured to use both Run-From-Package and AzureFiles. Run-From-Package will take precedence");
                     }
                     var blobContextApplied =
-                        await _runFromPackageHandler.ApplyBlobPackageContext(pkgContext, options.ScriptPath,
+                        await _runFromPackageHandler.ApplyRunFromPackageContext(pkgContext, options.ScriptPath,
                             azureFilesMounted, false);
 
                     if (!blobContextApplied && azureFilesMounted)
                     {
-                        _logger.LogWarning($"Failed to {nameof(_runFromPackageHandler.ApplyBlobPackageContext)}. Attempting to use local disk instead");
-                        await _runFromPackageHandler.ApplyBlobPackageContext(pkgContext, options.ScriptPath, false);
+                        _logger.LogWarning($"Failed to {nameof(_runFromPackageHandler.ApplyRunFromPackageContext)}. Attempting to use local disk instead");
+                        await _runFromPackageHandler.ApplyRunFromPackageContext(pkgContext, options.ScriptPath, false);
+                    }
+                }
+                else if (pkgContext.IsRunFromLocalPackage())
```

Comment:
>pkgContext.IsRunFromLocalPackage() [](http://example.com/codeflow?start=25&length=34)

since this can impact how IsFileSystemReadOnly is calculated
+ @pgombar for context and review.

Reply:
Hi @balag0 and @liliankasem, thanks for flagging. I think this new logic doesn't break anyhting related to IsFileSystemReadOnly because we return `false` if `WEBSITE_RUN_FROM_PACKAGE` app setting is present, which would be the case with this feature:
https://github.com/Azure/azure-functions-host/blob/2aae5b90962c6090a2f4799fe1944575b05f592e/src/WebJobs.Script.WebHost/Configuration/ScriptApplicationHostOptionsSetup.cs#L72-L91

 LGTM on that front.

Reply:
Thanks!